src/utils/utils.js

```diff
@@ -30,3 +30,11 @@ export var colorLuminance = function(hex, lum) {
 
   return rgb;
 };
+
+/*
+ * check if variable is function type. http://stackoverflow.com/questions/5999998/how-can-i-check-if-a-javascript-variable-is-function-type
+ */
+export var isFunction = function(functionToCheck) {
+    var getType = {};
+    return functionToCheck && getType.toString.call(functionToCheck) === '[object Function]';
```

Comment:
It looked like the most robust and it was checked as the accepted answer. I
can switch it to the shorter one if you'd prefer.

On Jul 31, 2016 5:05 PM, "Limon Monte" <notifications@github.com> wrote:

> In src/utils/utils.js
> <https://github.com/limonte/sweetalert2/pull/234#discussion_r72913938>:
>
> > @@ -30,3 +30,11 @@ export var colorLuminance = function(hex, lum) {
> >
> >    return rgb;
> >  };
> > +
> > +/*
> > + * check if variable is function type. http://stackoverflow.com/questions/5999998/how-can-i-check-if-a-javascript-variable-is-function-type
> > + */
> > +export var isFunction = function(functionToCheck) {
> > +    var getType = {};
> > +    return functionToCheck && getType.toString.call(functionToCheck) === '[object Function]';
>
> why did you pick this method instead if shorter if (typeof
> functionToCheck === "function") { (most popular answer in your link)?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/limonte/sweetalert2/pull/234/files/f8c61ab3c26c00351ede3a4d5c1b84681ab3fe06#r72913938>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADrdJN-iRDq8zDso-BftR_N8pqNSEktfks5qbThRgaJpZM4JZJoH>
> .
>